### PR TITLE
Added search with aggregation samples

### PR DESCRIPTION
**Pull request by araithel**

### CODE REVIEW

File: src/main/java/de/picturesafe/search/samples/aggregations/SearchWithAggregations.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    private void createTestRecords() {
+        final Instant now = Instant.now();
+        final Instant yesterday = now.minus(1, ChronoUnit.DAYS);
+        final Instant oneYearAgo = now.minus(365, ChronoUnit.DAYS);
+
+        // Insert 20 test records with city 'Hamburg', 'Tag A' and create date 'now'
+        LongStream.range(1, 21)
+                .forEach(id -> singleIndexElasticsearchService.addToIndex(DataChangeProcessingMode.BLOCKING,
```
Comment (jfrommann) [inline src/main/java/de/picturesafe/search/samples/aggregations/SearchWithAggregations.java:85]: A loop with single adds in blocking mode is a serious performance issue! I know this is no production code, but a sample should give a hint on how to use the API _correctly_.

File: src/main/java/de/picturesafe/search/samples/aggregations/SearchWithDateHistogramAggregation.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+    private void createTestRecords() {
+        final Instant now = Instant.now();
+        final Instant yesterday = now.minus(1, ChronoUnit.DAYS);
+        final Instant oneYearAgo = now.minus(365, ChronoUnit.DAYS);
+
+        // Insert 20 test records with create date 'now'
+        LongStream.range(1, 21)
+                .forEach(id -> singleIndexElasticsearchService.addToIndex(DataChangeProcessingMode.BLOCKING,
+                        DocumentBuilder.id(id).put("title", "This is a test title " + id).put("created", Date.from(now)).build()));
```
Comment (jfrommann) [inline src/main/java/de/picturesafe/search/samples/aggregations/SearchWithDateHistogramAggregation.java:84]: A loop with single adds in blocking mode is a serious performance issue! I know this is no production code, but a sample should give a hint on how to use the API correctly.

File: src/main/java/de/picturesafe/search/samples/aggregations/SearchWithAggregations.java [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+                        DocumentBuilder.id(id).put("title", "This is a test title " + id)
+                                .put("city", "London")
+                                .put("tag", "Tag B")
+                                .put("created", Date.from(yesterday)).build()));
+
+        // Insert 5 test records with city 'Paris', 'Tag A' and create date 'one year ago'
+        LongStream.range(31, 36)
+                .forEach(id -> singleIndexElasticsearchService.addToIndex(DataChangeProcessingMode.BLOCKING,
+                        DocumentBuilder.id(id).put("title", "This is a test title " + id)
+                                .put("city", "Paris").put("created", Date.from(oneYearAgo))
```
Comment (jfrommann) [inline src/main/java/de/picturesafe/search/samples/aggregations/SearchWithAggregations.java:104]: No new line as in the code above?

File: src/main/java/de/picturesafe/search/samples/aggregations/SearchWithTermsAggregation.java [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            showFacets(searchResult);
+        } finally {
+            singleIndexElasticsearchService.deleteIndexWithAlias();
+        }
+    }
+
+    private void createTestRecords() {
+        // Insert 20 test records with city 'Hamburg'
+        LongStream.range(1, 21)
+                .forEach(id -> singleIndexElasticsearchService.addToIndex(DataChangeProcessingMode.BLOCKING,
```
Comment (jfrommann) [inline src/main/java/de/picturesafe/search/samples/aggregations/SearchWithTermsAggregation.java:74]: A loop with single adds in blocking mode is a serious performance issue! I know this is no production code, but a sample should give a hint on how to use the API correctly.

File: src/main/java/de/picturesafe/search/samples/facet/Facet.java [hunk 5]
```diff
@@ -20,8 +36,14 @@
 
 import java.util.stream.LongStream;
 
+/**
+ * Aggregation/Facet sample.
+ *
+ * @deprecated Use samples of package {@link de.picturesafe.search.samples.aggregations} instead.
```
Comment (jfrommann) [inline src/main/java/de/picturesafe/search/samples/facet/Facet.java:42]: Just for curiosity: why are we keeping this sample?

Comment (araithel) [reply src/main/java/de/picturesafe/search/samples/facet/Facet.java:42]: It is still referenced in https://picturesafe-search.io/docs/reference/aggregations/.
After modification the sample can be removed.

File: src/main/java/de/picturesafe/search/samples/aggregations/SearchWithAggregations.java (hunk 1 above)
Comment (araithel) [reply src/main/java/de/picturesafe/search/samples/aggregations/SearchWithAggregations.java]: Test records are now added per bulk (background).

**Pull request merged**